permissions.go

```diff
@@ -0,0 +1,78 @@
+package stream_chat //nolint: golint
+
+import (
+	"net/http"
+	"path"
+	"time"
+)
+
+type CustomPermission struct {
```

Comment:
I don't like the name but Permission is already used by something else....

Reply:
`PermissionV2`.

Reply:
good idea